include/forcing/ForcingsEngineLumpedDataProvider.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <NGenConfig.h>
+
+#if NGEN_WITH_PYTHON
+
+#include <forcing/ForcingsEngineDataProvider.hpp>
+#include <forcing/DataProviderSelectors.hpp>
+
+namespace data_access {
+
+struct ForcingsEngineLumpedDataProvider final :
+  public ForcingsEngineDataProvider<double, CatchmentAggrDataSelector>
+{
+    using data_type      = double;
+    using selection_type = CatchmentAggrDataSelector;
```

Comment:
Rather than having these type aliases in each derived class, could you please add them to the base `DataProvider` class template?

E.g.

```diff
diff --git a/include/forcing/DataProvider.hpp b/include/forcing/DataProvider.hpp
index a8ffb5604..89da39f15 100644
--- a/include/forcing/DataProvider.hpp
+++ b/include/forcing/DataProvider.hpp
@@ -21,7 +21,7 @@ namespace data_access
             BACK_FILL
     };
 
-    template <class data_type, class selection_type> class DataProvider
+    template <class DataType, class SelectionType> class DataProvider
     {
         /** This class provides a generic interface to data services
         *
@@ -29,6 +29,9 @@ namespace data_access
 
         public:
 
+        using selection_type = SelectionType;
+        using data_type = DataType;
+
         virtual ~DataProvider() = default;
```